components/home/Challenge.tsx

```diff
@@ -153,14 +154,26 @@ export function Challenge({
                       );
                     }}
                   />
-                  <ThemedIconButton
-                    name="location-outline"
-                    onPress={() => {
-                      /* location button */
-                    }}
-                    size={25}
-                    color="white"
-                  />
+                  {challengeDB.location && (
+                    <ThemedIconButton
+                      name="location-outline"
+                      onPress={() => {
+                        navigation.navigate("MapScreen", {
+                          navigation: navigation,
+                          firestoreCtrl: firestoreCtrl,
+                          user: currentUser,
+                          location: {
+                            coords: {
+                              latitude: challengeDB.location?.latitude,
+                              longitude: challengeDB.location?.longitude,
+                            },
```

Comment:
Same here, why all the props ?

Reply:
I was not aware that we used `GeoPoint`s now and continued using `LocationObject`s. This is indeed way clearer and was modified accordingly!